source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,448 @@
+  * Port ID TLV (Type = 2)
+  * Port Description TLV (Type = 4), with a value of up to 2000 chars
+  * System Name TLV (Type = 5)
+  * System Description TLV (Type = 6)
+  * Management Address TLV (Type = 8)
+  * Port VLAN Id ( Type = 127, Oui = 0x0080c2 (802.1), Subtype = 1)
+  * VLAN name (Type = 127, Oui = 0x0080c2 (802.1), Subtype = 3)
+
+  has to be available on the oVirt Engine's REST API
+
+* Is there an upper limit of the acceptable age of the data?
+
+## Open Questions
+
+* How is a Port Description with a length than more than 512 chars represented
+  in LLDP?
+* Should oVirt Engine provide the age of the data?
+* Should oVirt Engine hide expired data?
+* Is there an upper limit of the acceptable age of the data?
+
+## Approach
+
+Steps to present the LLDP information:
+
+1. [Receive LLDP information on the Ethernet cable on the host](#vdsm-implementation-options)
+2. [Cache the LLDP information on the host](#vdsm-implementation-options)
+3. [Transport the LLDP information to oVirt Engine](#communication-of-vdsm-and-ovirt-engine)
+4. Present LLDP information:
+  * [On REST-API](#rest-api)
+  * [On Web-UI](#follow-up-features)
+  * [For oVirt Engine internal use](#follow-up-features)
+
+If the LLDP information would be used internally in oVirt Engine or in Web-UI, it
+would be useful to cache the information in the data base.
+
+But to meet the requirements of this feature, there is no need to store the
+information in the data base, because the information could be queried
+during the REST request is processed. This way the age of the information is
+kept minimal as possible.
+
+## Gathering the LLDP data from the hosts
+
+There are multiple alternatives for oVirt Engine to gather the LLDP data from
+the hosts:
+
+* Extend VDSM to provide the LLDP data
+
+* Use [Ansible's lldp module][3]
+
+Because the Ansible module is flagged as preview and requires lldpd, which is
+in only in EPEL repository for RHEL, VDSM has to be extended.
+
+## VDSM implementation options
+
+to be imported from [EdDev/ovirt-devwiki](https://github.com/EdDev/ovirt-devwiki/blob/master/design/network/lldp/lldp.md)
+
+## Communication of VDSM and oVirt Engine
+
+Since the LLDP information is required only sporadic, the new verb
+`Host.getNicNeighbor` is introduced.
+
+```yaml
+types:
+  Tlv: &Tlv
+    added: '4.2'
+    description: LLDP Information 
+    name: Tlv
+    properties:
+    - description: Human readable string to describe what the value is about.
+      name: name
+      type: string
+
+    - description: Structured information of the TLV
+      name: properties
+      type: *StringMap
+
+    - description: TLV type of IEEE 802.1AB.
+      name: type
+      type: string
+      datatype: uint
+
+    - description: organizationally unique identifier
+      name: oui
+      type: string
+      datatype: uint
+
+    - description: organizationally defined subtype
+      name: subtype
+      type: string
+      datatype: uint
+
+
+Host.getNicNeighbor:
```

Comment:
Neighbour is a bit generic, don't you think?
It is also used by iproute2 for managing arp entries (see man ip-neighbour), not exactly colliding but has the same issue.
Can we just call it lldpinfo or something like that?